CONTRIBUTING.md

```diff
@@ -0,0 +1,93 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make by creating an issue.
```

Comment:
Do you really want this? This can be awkward for smaller changes. Maybe make this a bit lighter, something like:
```suggestion
When contributing to this repository, you are welcome to submit a PR right away. To avoid disappointment, however, for larger changes it can be good to discuss it beforehand in a GitHub issue.
```